spine_engine/spine_engine.py

```diff
@@ -180,6 +173,54 @@ def __init__(
         self._event_stream = self._get_event_stream()
         self._multiprocess_manager = mp.Manager()
 
+    def _make_connections(self, connections):
+        """Returns a list of Connection instances based on given
+        serialized connections and permitted items. Creates Connection
+        instances only for connections that are coming from permitted items
+        or leaving from permitted items.
+
+        Args:
+            connections (list): Serialized connections in the DAG
+
+        Returns:
+            list: List of permitted Connections or an empty list if the DAG contains no connections
+        """
+        if not connections:
+            return list()
+        # List of item names that are permitted, i.e. selected for execution
+        permitted_items = [n for n, n_permitted in self._execution_permits.items() if n_permitted]
```

Comment:
Just a little note: on average, searching a specific item from `set` is faster than from `list` so `permitted_items` would be better off being a `set`. However, we don't care about a few nanoseconds here. Just something to perhaps keep in mind.